include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -2686,6 +2766,181 @@ MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::
   return cell_transfer.run(cell_transfer_test);
 }
 
+
+
+namespace RepartitioningPolicyTools
+{
+  namespace
+  {
+    template <typename MeshType>
+    unsigned int
+    compute_n_coarse_cells(const MeshType &mesh)
+    {
+      types::coarse_cell_id n_coarse_cells = 0;
+
+      for (const auto &cell : mesh.get_triangulation().active_cell_iterators())
+        if (!cell->is_artificial())
+          n_coarse_cells =
+            std::max(n_coarse_cells, cell->id().get_coarse_cell_id());
+
+      return Utilities::MPI::max(n_coarse_cells, mesh.get_communicator()) + 1;
+    }
+
+
+
+    template <typename MeshType>
+    unsigned int
+    compute_n_global_levels(const MeshType &mesh)
+    {
+      return mesh.get_triangulation().n_global_levels();
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    add_indices(const TriaIterator<CellAccessor<dim, spacedim>> &cell,
+                const internal::CellIDTranslator<dim> &cell_id_translator,
+                IndexSet &                             is_fine)
+    {
+      is_fine.add_index(cell_id_translator.translate(cell));
+
+      if (cell->level() > 0 &&
+          (cell->index() % Utilities::pow<unsigned int>(2, dim)) == 0)
```

Comment:
Isn't this hardcoding the cell type to quad/hex here? Also, the code is a bit confusing, because this hardcodes the decision to the first-child rule (otherwise the resulting index set would be overlapping between the MPI processes), so I think this should be better marked as part of that strategy.

Reply:
OK. I have added an assert that this is only working for hex-meshes at the moment (although for simplex meshes it should be the same). I have also renamed the function so that it is more clear that it is related to the given policy.